File: virtual-library/ios/VirtualLibraryTests/Models/LibraryTests.swift
import XCTest
@testable import VirtualLibrary

final class LibraryTests: XCTestCase {
    
    func testLibraryInitialization() {
        // Arrange
        let id = "lib-123"
        let name = "My Library"
        let description = "Test library"
        let owner = "test@example.com"
        
        // Act
        let library = Library(
            id: id,
            name: name,
            description: description,
            owner: owner,
            books: []
        )
        
        // Assert
        XCTAssertEqual(library.id, id)
        XCTAssertEqual(library.name, name)
        XCTAssertEqual(library.description, description)
        XCTAssertEqual(library.owner, owner)
        XCTAssertTrue(library.books.isEmpty)
    }
    
    func testLibraryEquality() {
        // Arrange
        let library1 = Library(id: "1", name: "Lib", description: "", owner: "owner", books: [])
        let library2 = Library(id: "1", name: "Lib", description: "", owner: "owner", books: [])
        let library3 = Library(id: "2", name: "Lib", description: "", owner: "owner", books: [])
        
        // Assert
        XCTAssertEqual(library1, library2)
        XCTAssertNotEqual(library1, library3)
    }
}

File: virtual-library/api/VirtualLibrary.Api.Tests/Domain/LibraryTests.cs
using FluentAssertions;
using VirtualLibrary.Api.Domain;
using Xunit;

namespace VirtualLibrary.Api.Tests.Domain;

public class LibraryTests
{
    [Fact]
    public void Library_Initialization_SetsProperties()
    {
        // Arrange & Act
        var library = new Library
        {
            Id = Guid.NewGuid(),
            Name = "My Library",
            Description = "Test Description",
            Owner = "test@example.com",
            UserId = Guid.NewGuid()
        };

        // Assert
        library.Name.Should().Be("My Library");
        library.Description.Should().Be("Test Description");
        library.Owner.Should().Be("test@example.com");
        library.BookIds.Should().BeEmpty();
    }

    [Fact]
    public void Library_CanAddBookIds()
    {
        // Arrange
        var library = new Library
        {
            Id = Guid.NewGuid(),
            Name = "Test Library",
            Owner = "test@example.com",
            UserId = Guid.NewGuid()
        };

        var bookId = Guid.NewGuid();

        // Act
        library.BookIds.Add(bookId);

        // Assert
        library.BookIds.Should().HaveCount(1);
        library.BookIds.First().Should().Be(bookId);
    }

    [Fact]
    public void Library_Tags_CanBeModified()
    {
        // Arrange
        var library = new Library
        {
            Id = Guid.NewGuid(),
            Name = "Test",
            UserId = Guid.NewGuid()
        };

        // Act
        library.Tags.Add("fiction");
        library.Tags.Add("classics");

        // Assert
        library.Tags.Should().HaveCount(2);
    }
}
